ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Examples.hs

```diff
@@ -230,13 +264,33 @@ fromShelleyLedgerExamplesPraos ShelleyLedgerExamples {
                                   }
     , shelleyLedgerState      = sleNewEpochState
     , shelleyLedgerTransition = ShelleyTransitionInfo {shelleyAfterVoting = 0}
+    , shelleyLedgerTables     = emptyLedgerTables
     }
     chainDepState = translateChainDepState @(TPraos (EraCrypto era)) @(Praos (EraCrypto era))
       $ TPraosState (NotOrigin 1) sleChainDepState
     extLedgerState = ExtLedgerState
                        ledgerState
                        (genesisHeaderState chainDepState)
+    ledgerTables = ShelleyLedgerTables
+                 $ ValuesMK
+                 $ Map.fromList
+                 $ zip exampleTxIns exampleTxOuts
+      where
+        exampleTxIns :: [TxIn (EraCrypto era)]
+        exampleTxIns  =
+          case toList $ getAllTxInputs (sleTx ^. LC.bodyTxL) of
+            [] -> error "No transaction inputs were provided to construct the ledger tables"
+                  -- We require at least one transaction input (and one
+                  -- transaction output) in the example provided by
+                  -- cardano-ledger to make sure that we test the serialization
+                  -- of ledger tables with at least one non-trivial example.
+            xs -> xs
 
+        exampleTxOuts :: [LC.TxOut era]
+        exampleTxOuts =
+          case toList (sleTx ^. LC.bodyTxL ^. LC.outputsTxBodyL) of
+            [] -> error "No transaction outputs were provided to construct the ledger tables"
+            xs -> xs
```

Comment:
This is duplicated code -- maybe factor out a `mkLedgerTables` function from `fromShelleyLedgerExamplesPraos` and `fromShelleyLedgerExamples`?